src/main/host/descriptor/tcp.c

```diff
@@ -1325,7 +1329,13 @@ static void _tcp_flush(TCP* tcp, const Host* host) {
         PacketTCPHeader* header = packet_getTCPHeader(packet);
 
         _rswlog(tcp, "I just received packet %d\n", header->sequence);
-        if(header->sequence == tcp->receive.next) {
+        if (header->sequence < tcp->receive.next) {
+            // This is a (probably retransmitted) copy of a packet we already stored
+            // and delivered to the plugin.
+            trace("Removing packet %u with duplicate data", header->sequence);
+            priorityqueue_pop(tcp->unorderedInput);
+            packet_unref(packet);
+        } else if (header->sequence == tcp->receive.next) {
             /* move from the unordered buffer to user input buffer */
             gboolean fitInBuffer = legacysocket_addToInputBuffer(&(tcp->super), host, packet);
```

Comment:
aside: Below the code does a `packet_unref(packet)` before the `packet_getPayloadSize(packet)`. Maybe while you're editing this code, reorder these lines so that we don't use the packet after we unref it?

Reply:
Will do :)